source/common/runtime/runtime_features.cc

```diff
@@ -45,6 +45,41 @@ constexpr const char* disallowed_features[] = {
     // Acts as both a test entry for deprecated.proto and a marker for the Envoy
     // deprecation scripts.
     "envoy.deprecated_features.deprecated.proto:is_deprecated_fatal",
+    // 1.12.0
+    "envoy.deprecated_features.route.proto:regex_match",
+    "envoy.deprecated_features.route.proto:regex",
+    "envoy.deprecated_features.accesslog.proto:config",
+    "envoy.deprecated_features.thrift_proxy.proto:config",
+    "envoy.deprecated_features.cds.proto:tls_context",
+    "envoy.deprecated_features.route.proto:allow_origin",
+    "envoy.deprecated_features.string.proto:regex",
+    "envoy.deprecated_features.grpc_service.proto:config",
+    "envoy.deprecated_features.tcp_proxy.proto:deprecated",
+    "envoy.deprecated_features.overload.proto:config",
+    "envoy.deprecated_features.route.proto:value",
+    "envoy.deprecated_features.base.proto:config",
+    "envoy.deprecated_features.http_connection_manager.proto:idle_timeout",
+    "envoy.deprecated_features.lds.proto:use_original_dst",
+    "envoy.deprecated_features.config_source.proto:DEPRECATED_AND_UNAVAILABLE_DO_NOT_USE",
+    "envoy.deprecated_features.cds.proto:config",
+    "envoy.deprecated_features.route.proto:pattern",
+    "envoy.deprecated_features.listener.proto:tls_context",
+    "envoy.deprecated_features.listener.proto:config",
+    "envoy.deprecated_features.health_check.proto:config",
+    "envoy.deprecated_features.http_connection_manager.proto:[(validate.rules).enum",
```

Comment:
?

Reply:
mmm, that's
    OperationName operation_name = 1
        [(validate.rules).enum = {defined_only: true}, deprecated = true];

Again I think there's some things that are beyond simple checking - we'd have to either do this programatically which would be awesome but I'm not sure how to do, or accept some clean up

Reply:
See above comment, let's sync up with @htuch